live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/TypeFilterApplier.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.hxi_connector.live_ingester.adapters.messaging.repository.filter;
+
+import java.util.List;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import org.alfresco.hxi_connector.live_ingester.adapters.config.properties.Filter;
+import org.alfresco.repo.event.v1.model.DataAttributes;
+import org.alfresco.repo.event.v1.model.NodeResource;
+import org.alfresco.repo.event.v1.model.RepoEvent;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class TypeFilterApplier implements NodeFilterApplier
+{
+    @Override
+    public boolean applyFilter(RepoEvent<DataAttributes<NodeResource>> repoEvent, Filter filter)
+    {
+        final String nodeType = repoEvent.getData().getResource().getNodeType();
+        final List<String> allowed = filter.type().allow();
+        final List<String> denied = filter.type().deny();
+        log.atDebug().log("Applying type filters on repo event of id: {}. Event node type: {}. Allowed types: {}. Denied types: {}", repoEvent.getId(), nodeType, allowed, denied);
+        final boolean allow = filterAllowed(nodeType, allowed);
+        final boolean deny = filterDenied(nodeType, denied);
```

Comment:
I would get rid of these `final` keywords, imo they are unnecessary

Reply:
But these variables are `final`... I don't get why wouldn't I "mark" them final then...

Reply:
idk it is just an opinion but for me it makes unnecessary noise - you can clearly see that you dont reassign any of these variables (even ide will highlight variables that are reassigned)

Reply:
I've been against marking all variables/method params `final` for a long time especially that at some point IntelliJ's `Extract to variable` started to do that out of the box (it lasted quite for short time). Then I got used to it and started seeing more values in having a habit to that for each variable/param. It sometimes can make people think twice when you try to re-assign and then your IDE goes red - without `final` it doesn't. And believe some people only start re-thinking when IDE goes red.
From my experience it does no harm but can do good, so sorry - but we seem to have different opinions.

Reply:
yep but at the end of the day, it is all just about consistency - we should always add or always skip it ;) Closing the conversation but imo it is something that is worth to standardise